src/v/datalake/tests/proto_to_arrow_gtest.cc

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+#include "datalake/errors.h"
+#include "datalake/protobuf_to_arrow_converter.h"
+#include "datalake/tests/proto_to_arrow_test_utils.h"
+#include "test_utils/test.h"
+
+#include <google/protobuf/descriptor.h>
+#include <google/protobuf/message.h>
+#include <gtest/gtest.h>
+
+#include <stdexcept>
+#include <string>
+
+namespace datalake {
+TEST(ArrowWriter, EmptyMessageTest) {
+    using namespace datalake;
+    test_data test_data;
+    std::string schema = test_data.empty_schema;
+
+    proto_to_arrow_converter converter(schema);
+
+    std::string serialized_message = generate_empty_message();
+
+    auto parsed_message = converter.parse_message(serialized_message);
+    EXPECT_NE(parsed_message, nullptr);
+    EXPECT_EQ(parsed_message->GetTypeName(), "datalake.proto.empty_message");
```

Comment:
Should we also validate the contents of the message?

Reply:
This is an empty message. I added validation code to the SimpleMessageTest below.